File: two_sum/standard.rb
def two_sum(ints, s)
    # create a history hash
    # this object will be used for its key-value pairs.
    # the key will be the difference and the value will be the index
    history = Hash.new

    # iterate through the integers array
    ints.each do |i|
      if history[i] != nil
        # if a key of int exists in our history, a pair exists
        # return an array with the original index first and the curretn index second
        return [history[i], i]
      else
        # if no key exists get the difference of the sum and index
        difference = s - i

        # create a key-value pair with the difference as the key
        # you want the difference to be the key because is what should be appearing later in the array
        # set the value to the index
        history[difference] = i
      end
    end
    
    return "No Pair Sum"
end

File: two_sum/standard.js
function two_sum(ints, sum){
    // create a history object
    // we will use this object for storing indices and their differences with the sum argument
    let history = {}

    // iterate through the array
    for (let i = 0; i < ints.length; i++){
      if (history[ints[i]] !== undefined){
        // if an integer key is found a pair exists
        // return an array with the original key first, and second key second
        return [history[ints[i]], ints[i]]
      } else {
        // if no key of index exists, find the difference between the sum and index
        let difference = sum - ints[i]

        // set key of difference
        // you want to set a key of difference 
        history[difference] = ints[i]
      }
    }

    // if no pair is found, return the string
    return "No Pair Sum"
}